crates/polars-core/src/datatypes/dtype.rs

```diff
@@ -320,6 +320,57 @@ impl DataType {
         }
     }
 
+    /// Return whether the cast to `to` makes sense.
+    ///
+    /// This will never be precise but gives an initial estimate.
+    pub fn can_cast_to(&self, to: &DataType) -> bool {
```

Comment:
Can this return `Option<bool>`? On `None` we don't know and we must continue. But on `Some<false>` we could already error early.